### DISCUSSION TITLE
Improving Building static HTML for pages speeds

### DISCUSSION THREAD

**Discussion by kdichev** [gatsbyjs/gatsby#31844] [Help]
Hi guys, I am trying to figure out a way to improve/debug the speed of the `Building static HTML for pages` step but I am currently getting inconsistent results on that, so I am wondering what the cause of that would be!?

We have two self-hosted EC2 ubuntu build agents with 16 cores:
Staging Agent (used for PR and Staging builds)
Production Agent (used for Production builds)

our last staging build (Staging Agent) finished with 150 pages per second:
```
success Building static HTML for pages - 99.927s - 15056/15056 150.67/s
```

and the same commit build in production (Production Agent) finished with 72 pages per second:
```
success Building static HTML for pages - 208.598s - 15090/15090 72.34/s
```

interestingly enough when I run a full PR build (Staging Agent)  it finishes with 69 pages per second:
```
success Building static HTML for pages - 215.471s - 15055/15055 69.87/s
```

I am wondering what the cause of that would be?

In the perfect scenario we would like to get 150 pages per second consistently and be able to improve it to let's say 300 page per second!

So I guess my question here is how to go about debugging this step and what actions we can take to improve it?

**Comment by KyleAMathews** [comment]
Huh that is odd.

What comes to mind to check is there could be other services competing for resources so run `top` or something during a build to see if there's other processes using CPU/memory/IO.

Build speeds for SSR improve pretty linearly with system resources so you could also try a 32 or 64 core box.

**Reply by kdichev** [reply]
those agents are used only for the build and are setup in ado so I don't think there are other resources competing but I will check. What I was thinking also is to re create those instances and see if their age is of some kind of effect (I think they are 6+ months old vms). 

I know the speeds will improve with 32/64 core boxes but those are very expensive to run
